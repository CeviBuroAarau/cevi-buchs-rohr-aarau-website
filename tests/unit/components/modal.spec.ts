import { shallowMount, Wrapper } from "@vue/test-utils";
import Modal from "@/components/modal.vue";

describe("Modal Component", () => {
  test("render success", () => {
    const wrapper = shallowMount(Modal, {
      stubs: [],
      propsData: {
        title: "Titel",
        message: "Message",
        type: "success",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test("renders error", () => {
    const wrapper = shallowMount(Modal, {
      stubs: [],
      propsData: {
        title: "Titel",
        message: "Message",
        type: "error",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test("show", async () => {
    const wrapper: Wrapper<Modal & { [key: string]: any }> = shallowMount(
      Modal,
      {
        stubs: [],
        propsData: {
          title: "Titel",
          message: "Message",
          type: "error",
        },
      }
    );

    await wrapper.vm.open();

    const container = wrapper.find("div.is-active");
    expect(container.exists()).toBe(true);
  });

  test("close", async () => {
    const wrapper: Wrapper<Modal & { [key: string]: any }> = shallowMount(
      Modal,
      {
        stubs: [],
        propsData: {
          title: "Titel",
          message: "Message",
          type: "error",
        },
      }
    );

    await wrapper.vm.close();

    const container = wrapper.find("div.is-active");
    expect(container.exists()).toBe(false);
  });
});
