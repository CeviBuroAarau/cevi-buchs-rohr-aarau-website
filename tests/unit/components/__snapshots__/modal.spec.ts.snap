// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render success 1`] = `
<div
  class="modal"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Titel
      </p>
    </header>
     
    <section
      class="modal-card-body"
    >
      <div
        class="notification"
      >
        
        Message
      
      </div>
    </section>
     
    <footer
      class="modal-card-foot"
    >
      <button
        class="button is-success"
      >
        OK
      </button>
    </footer>
  </div>
</div>
`;

exports[`renders error 1`] = `
<div
  class="modal"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Titel
      </p>
    </header>
     
    <section
      class="modal-card-body"
    >
      <div
        class="notification is-danger"
      >
        
        Message
      
      </div>
    </section>
     
    <footer
      class="modal-card-foot"
    >
      <button
        class="button is-success"
      >
        OK
      </button>
    </footer>
  </div>
</div>
`;
